dbms/src/Operators/CoprocessorReaderSourceOp.cpp

```diff
@@ -0,0 +1,122 @@
+// Copyright 2023 PingCAP, Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <DataStreams/IBlockInputStream.h>
+#include <Flash/Coprocessor/GenSchemaAndColumn.h>
+#include <Operators/CoprocessorReaderSourceOp.h>
+
+namespace DB
+{
+CoprocessorReaderSourceOp::CoprocessorReaderSourceOp(
+    PipelineExecutorStatus & exec_status_,
+    const String & req_id,
+    CoprocessorReaderPtr coprocessor_reader_)
+    : SourceOp(exec_status_, req_id)
+    , coprocessor_reader(coprocessor_reader_)
+{
+    assert(coprocessor_reader);
+    setHeader(Block(getColumnWithTypeAndName(toNamesAndTypes(coprocessor_reader->getOutputSchema()))));
+}
+
+void CoprocessorReaderSourceOp::operatePrefix()
+{
+    coprocessor_reader->open();
+}
+
+void CoprocessorReaderSourceOp::operateSuffix()
+{
+    LOG_INFO(log, "finish read {} rows from remote coprocessor", total_rows);
```

Comment:
```suggestion
    LOG_DEBUG(log, "finish read {} rows from remote coprocessor", total_rows);
```
ref https://github.com/pingcap/tiflash/pull/7285/files#diff-c331695ee7c42569e94d53133f5c208892e62511d2168527838fe19d5bbbd8c5R173